src/transaction/components/TransactionProvider.tsx

```diff
@@ -39,21 +40,49 @@ export function TransactionProvider({
   const [gasFee, setGasFee] = useState('');
   const [isToastVisible, setIsToastVisible] = useState(false);
 
-  const { status, writeContracts } = useWriteContracts({
+  const { status: statusWriteContracts, writeContracts } = useWriteContracts({
+    onError,
+    setErrorMessage,
+    setTransactionId,
+  });
+
+  const {
+    status: statusWriteContract,
+    writeContract,
+    data: writeContractTransactionHash,
+  } = useWriteContract({
     onError,
     setErrorMessage,
     setTransactionId,
   });
 
   const { transactionHash } = useCallsStatus({ onError, transactionId });
 
-  const handleSubmit = useCallback(() => {
+  const fallbackToWriteContract = useCallback(async () => {
+    for (const contract of contracts) {
+      try {
+        await writeContract(contract);
+      } catch (_err) {
+        setErrorMessage('genericErrorMessage');
```

Comment:
Following the pattern in the `Transaction Component`. I believe that error handling will be addresses in a fast follow